src/libraries/System.Runtime/tests/System.Reflection.Tests/AssemblyTests.cs

```diff
@@ -181,6 +181,19 @@ public void GetEntryAssembly()
             Assert.True(correct, $"Unexpected assembly name {assembly}");
         }
 
+        [Fact]
+        public void SetEntryAssembly()
+        {
+            Assembly? originalAssembly = Assembly.GetEntryAssembly();
+            Assert.NotNull(originalAssembly);
+
+            Assembly.SetEntryAssembly(null);
+            Assert.Null(Assembly.GetEntryAssembly());
+
+            Assembly.SetEntryAssembly(originalAssembly);
```

Comment:
```suggestion
            Assembly.SetEntryAssembly(typeof(AssemblyTests).Assembly);
```
This may be more interesting test case.